modules/persistent-storage-csi-vsphere-change-max-snapshot.adoc

```diff
@@ -0,0 +1,128 @@
+// Module included in the following assemblies:
+//
+// * storage/persistent_storage/persistent-storage-vsphere.adoc
+// * storage/persistent_storage/persistent-storage-csi-snapshots.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="vsphere-change-max-snapshot_{context}"]
+= Changing the maximum number of snapshots for vSphere
+
+The default maximum number of snapshots per volume in vSphere Container Storage Interface (CSI) is 3. You can change the maximum number up to 32 per volume. 
+
+However, be aware that increasing the snapshot maximum involves a performance trade off, so for better performance use only 2 to 3 snapshots.
+
+For more VMWare snapshot performance recommendations, see *_Additional Resources_*.
+
+.Prerequisites
+
+* Access to the cluster with administrator rights.
+
+.Procedure
+
+. Check the current config map by the running the following command:
++
+[source, terminal]
+----
+$ oc -n openshift-cluster-csi-drivers get cm/vsphere-csi-config -o yaml
+----
++
+.Example output
++
+[source, terminal]
+----
+apiVersion: v1
+data:
+  cloud.conf: |+
+    # Labels with topology values are added dynamically via operator
+    [Global]
+    cluster-id = rbednar-vsphere-01-cwv8p
+
+    [VirtualCenter "vcenter.devqe.ibmc.devcluster.openshift.com"]
+    insecure-flag           = true
+    datacenters             = DEVQEdatacenter
+    migration-datastore-url = ds:///vmfs/volumes/vsan:527320283a8c3163-2faa6dc5949a3a28/
+
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2024-03-06T09:46:40Z"
+  name: vsphere-csi-config
+  namespace: openshift-cluster-csi-drivers
+  resourceVersion: "126687"
+----
+
+. Change the snapshot limit by running the following command:
++
+* Set *global* snapshot limit:
++
+[source, terminal]
+----
+$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"globalMaxSnapshotsPerBlockVolume": 10}}}}'
+clustercsidriver.operator.openshift.io/csi.vsphere.vmware.com patched
+----
++
+In this example, the global limit is being changed to 10 (`globalMaxSnapshotsPerBlockVolume` set to 10).
+
+* Set *Virtual Volume* snapshot limit:
++
+[source, terminal]
+----
+$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"GranularMaxSnapshotsPerBlockVolumeVvol": 5}}}}'
+clustercsidriver.operator.openshift.io/csi.vsphere.vmware.com patched
+----
++
+In this example, the Virtual Volume limit is being changed to 5 (`GranularMaxSnapshotsPerBlockVolumeVvol` set to 5).
+
+* Set *vSAN* snapshot limit:
++
+[source, terminal]
+----
+$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"GranularMaxSnapshotsPerBlockVolumeVsan": 7}}}}'
```

Comment:
```suggestion
$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"granularMaxSnapshotsPerBlockVolumeInVSAN": 7}}}}'
```
Seems it is not accture enough for the API parameter names.
```console
$ oc explain clustercsidriver.spec.driverConfig.vSphere
KIND:     ClusterCSIDriver
VERSION:  operator.openshift.io/v1

RESOURCE: vSphere <Object>

DESCRIPTION:
     vsphere is used to configure the vsphere CSI driver.

FIELDS:
   globalMaxSnapshotsPerBlockVolume	<integer>
     globalMaxSnapshotsPerBlockVolume is a global configuration parameter that
     applies to volumes on all kinds of datastores. If unset it defaults to 3.
     Increasing number of snapshots above 3 can have negative impact on
     performance, for more details see: https://kb.vmware.com/s/article/1025279
     Volume snapshot documentation:
     https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/3.0/vmware-vsphere-csp-getting-started/GUID-E0B41C69-7EEB-450F-A73D-5FD2FF39E891.html

   granularMaxSnapshotsPerBlockVolumeInVSAN	<integer>
     granularMaxSnapshotsPerBlockVolumeInVSAN is a granular configuration
     parameter on vSAN datastore only. It overrides
     GlobalMaxSnapshotsPerBlockVolume if set, while it falls back to the global
     constraint if unset.

   granularMaxSnapshotsPerBlockVolumeInVVOL	<integer>
     granularMaxSnapshotsPerBlockVolumeInVVOL is a granular configuration
     parameter on Virtual Volumes datastore only. It overrides
     GlobalMaxSnapshotsPerBlockVolume if set, while it falls back to the global
     constraint if unset.

   topologyCategories	<[]string>
     topologyCategories indicates tag categories with which vcenter resources
     such as hostcluster or datacenter were tagged with. If cluster
     Infrastructure object has a topology, values specified in Infrastructure
     object will be used and modifications to topologyCategories will be
     rejected.
```

Reply:
Done.